sdk/formrecognizer/ai-form-recognizer/src/lro/util/poller.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PollerLike, PollOperationState } from "@azure/core-lro";
+import { delayMs } from "./delayMs";
+
+const DEFAULT_POLLING_INTERVAL = 5000;
+
+/**
+ * A specification for a long-running operation, which defines the poller flow.
+ * @internal
+ */
+export interface OperationSpec<TState extends PollOperationState<unknown>> {
+  /**
+   * A function that produces the first operation state for this LRO.
+   */
+  init: () => Promise<TState>;
+  /**
+   * A function that consumes the existing state of the poller and produces the next state.
+   */
+  poll: (state: TState) => Promise<TState>;
+  /**
+   * A function that serializes the state into a string.
+   */
+  serialize: (state: TState) => string;
+}
+
+/**
+ * Uniform poller implementation, creates a poller based on a PollerSpec.
+ *
+ * @internal
+ */
+export async function lro<TResult, TState extends PollOperationState<TResult>>(
```

Comment:
Please make sure you have 100% test coverage for it.

Reply:
I will add an issue to cover all of the functionality here. We're over 80% line coverage in this file and are testing everything we tested in the previous SDK.